### PR TITLE
Added User Guide

### PR DESCRIPTION
**Pull request by naveentatikonda**
Signed-off-by: Naveen Tatikonda <navtat@amazon.com>

### Description
Added USER_GUIDE and link to USER_GUIDE from README

### Issues Resolved
https://github.com/opensearch-project/logstash-output-opensearch/issues/79

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    opensearch {        
+          hosts  => ["https://hostname:port"]     
+          auth_type => {            
+              type => 'basic'           
+              user => 'admin'           
+              password => 'admin'           
+          }             
+          index => "logstash-logs-%{+YYYY.MM.dd}"       
+   }            
+}
```
Comment (VijayanB) [inline USER_GUIDE.md:50]: Move this first and AWS IAM below.

Comment (naveentatikonda) [reply USER_GUIDE.md]: updated

**Comment by aetter** [comment]
The content looks good (didn't know we were adding IAM auth), but I'd rather the sample configurations and parameter descriptions just go here: https://opensearch.org/docs/clients/logstash/ship-to-opensearch/#opensearch-output-plugin

Any particular reason we want it in this repo?

**Comment by naveentatikonda** [comment]
> The content looks good (didn't know we were adding IAM auth), but I'd rather the sample configurations and parameter descriptions just go here: https://opensearch.org/docs/clients/logstash/ship-to-opensearch/#opensearch-output-plugin
> 
> Source page: https://github.com/opensearch-project/documentation-website/blob/main/_clients/logstash/ship-to-opensearch.md
> 
> Any particular reason we want it in this repo?

Got it. I will update it over there. Instead of maintaining two repos, logstash-output-amazon_es and logstash-output-opensearch, we thought to have one repo(logstash-output-opensearch) for both open source and managed service.

**Comment by naveentatikonda** [comment]
Added in documentation-website
https://github.com/opensearch-project/documentation-website/pull/184

**Pull request closed without merging**